File: src/ios/Sources/AnsightCore/PairingWifiPreflight.swift
import Foundation

#if canImport(Network)
import Network
#endif

enum PairingWifiPreflight {
    static func getStatus() -> PairingWifiPreflightStatus {
        #if canImport(Network)
        let path = currentNetworkPathSnapshot()
        guard path.status == .satisfied else {
            return .notConnected
        }

        if path.usesInterfaceType(.wifi) || path.usesInterfaceType(.wiredEthernet) {
            return .connected
        }

        if path.usesInterfaceType(.cellular) {
            return .cellular
        }

        return .unknown
        #else
        return .unknown
        #endif
    }

    #if canImport(Network)
    private static func currentNetworkPathSnapshot() -> NWPath {
        let monitor = NWPathMonitor()
        let queue = DispatchQueue(label: "ai.ansight.pairing.wifi-preflight")
        monitor.start(queue: queue)
        Thread.sleep(forTimeInterval: 0.05)
        monitor.cancel()
        return monitor.currentPath
    }
    #endif
}

File: src/dotnet/Ansight.Core/Pairing/PairingWifiPreflight.cs
namespace Ansight.Pairing;

internal enum PairingWifiPreflightStatus
{
    Unknown = 0,
    Connected = 1,
    NotConnected = 2,
    Cellular = 3
}

internal static partial class PairingWifiPreflight
{
    public static PairingWifiPreflightStatus GetStatus()
    {
        try
        {
            return GetPlatformStatusCore();
        }
        catch (Exception ex)
        {
            global::Ansight.Logger.Info($"Wi-Fi preflight could not determine current network status: {ex.Message}");
            return PairingWifiPreflightStatus.Unknown;
        }
    }

    private static partial PairingWifiPreflightStatus GetPlatformStatusCore();
}
